default.nix

```diff
@@ -16,21 +16,24 @@ let
     overlays = [ ];
     inherit system;
   };
-
+  nix-dev-python-pkgs = with pkgs.python310.pkgs; [
+    linkify-it-py
+    myst-parser
+    sphinx
+    sphinx-book-theme
+    sphinx-copybutton
+    sphinx-design
+    sphinx-notfound-page
+    sphinx-sitemap
+    pkgs.perl
+  ];
   nix-dev =
     pkgs.stdenv.mkDerivation {
       name = "nix-dev";
       src = ./.;
-      nativeBuildInputs = with pkgs.python310.pkgs; [
-        linkify-it-py
-        myst-parser
-        sphinx
-        sphinx-book-theme
-        sphinx-copybutton
-        sphinx-design
-        sphinx-notfound-page
-        sphinx-sitemap
-        pkgs.perl
+      nativeBuildInputs = [
+        nix-dev-python-pkgs
+        pkgs.texlive.combined.scheme-full
```

Comment:
Indeed downloading multiple gigabyte for a single pdf on each ci run is too slow.
https://github.com/rgri/tex2nix/ could help to cut it down.